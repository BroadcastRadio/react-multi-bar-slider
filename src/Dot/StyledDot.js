import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import transition from '../utils/transition';

const StyledDot = styled('span')({
  position: 'absolute',
  right: 0,
  display: 'block',
  zIndex: 5,
  borderRadius: '50%'
}, ({ hasIcon, width, height, color, reversed, noTransition, css }) => ({
  top: hasIcon ? 0 : '50%',
  left: reversed ? 0 : 'auto',
  right: reversed ? 'auto' : 0,
  transform: hasIcon
    ? 'translateX(-50%)'
    : reversed ? 'translate(-50%, -50%)' : 'translate(50%, -50%)',
  width: hasIcon ? 0 : width,
  height: hasIcon ? 0 : height,
  backgroundColor: hasIcon ? 'transparent' : color,
  transition: noTransition ? 'none' : transition,
  ...css
}));

StyledDot.propTypes = {
  hasIcon: PropTypes.bool.isRequired,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  color: PropTypes.string.isRequired,
  reversed: PropTypes.bool.isRequired,
  noTransition: PropTypes.bool.isRequired,
  css: PropTypes.object
};

StyledDot.defaultProps = {
  width: 28,
  height: 28,
  css: null
};

export default StyledDot;
