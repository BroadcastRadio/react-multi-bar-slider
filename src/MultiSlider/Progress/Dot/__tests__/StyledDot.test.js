import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import StyledDot from '../StyledDot';

Enzyme.configure({ adapter: new Adapter() });

beforeAll(() => {
  global.styledDotProps = {
    hasIcon: false,
    width: 30,
    height: 30,
    color: 'green',
    reversed: false,
    noTransition: false
  };
});

afterAll(() => {
  delete global.styledDotProps;
});

describe('StyledDot.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<StyledDot {...styledDotProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const styledDot = render(<StyledDot {...styledDotProps} />);
    expect(styledDot).toBeDefined();
  });
});
