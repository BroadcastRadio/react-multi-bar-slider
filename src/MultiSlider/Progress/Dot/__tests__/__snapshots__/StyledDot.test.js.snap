// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledDot.js matches the snapshot 1`] = `
<span
  className="css-154r8lw"
/>
`;
